tests/nosetests/pyanaconda_tests/module_source_url_test.py

```diff
@@ -30,6 +37,70 @@ def setUp(self):
         self.url_source_module = URLSourceModule()
         self.url_source_interface = URLSourceInterface(self.url_source_module)
 
+    def _check_dbus_property(self, *args, **kwargs):
+        check_dbus_property(
+            self,
+            PAYLOAD_SOURCE_URL,
+            self.url_source_interface,
+            *args, **kwargs
+        )
+
     def type_test(self):
         """Test URL source has a correct type specified."""
         self.assertEqual(SourceType.URL.value, self.url_source_interface.Type)
+
+    def set_url_base_source_properties_test(self):
+        data = URLData()
+        data.url = "http://example.com/repo"
+        data.type = URL_TYPE_BASEURL
+
+        self._check_dbus_property(
+            "URL",
+            URLData.to_structure(data)
+        )
+
+    def set_url_mirrorlist_properties_test(self):
+        data = URLData()
+        data.url = "http://forthehorde.com/mirrorlist?url"
+        data.type = URL_TYPE_MIRRORLIST
+
+        self._check_dbus_property(
+            "URL",
+            URLData.to_structure(data)
+        )
+
+    def set_url_metalink_properties_test(self):
+        data = URLData()
+        data.url = "https://alianceFTW/metalink?nopesir"
+        data.type = URL_TYPE_METALINK
+
+        self._check_dbus_property(
+            "URL",
+            URLData.to_structure(data)
+        )
+
+    def set_empty_url_properties_test(self):
+        self._check_dbus_property(
+            "URL",
+            URLData.to_structure(URLData())
+        )
+
+    def set_invalid_url_type_properties_test(self):
+        data = URLData()
+        data.url = "http://test"
+        data.type = "DOES-NOT-EXISTS"
+
+        with self.assertRaises(InvalidValueError):
+            self._check_dbus_property(
+                "URL",
+                URLData.to_structure(data)
+            )
+
+    def set_raw_url_with_properties_test(self):
+        data = {
+            "url": get_variant(Str, "http://NaNaNaNaNaNa/Batmaaan"),
+            "type": get_variant(Str, URL_TYPE_METALINK)
+        }
+        self._check_dbus_property(
+            "URL",
+            data)
```

Comment:
I'd move the ')' to the next line for consistency and beauty.

Reply:
Fixed.